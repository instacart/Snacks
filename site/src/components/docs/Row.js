import React, {Fragment} from 'react'
import Playground from '../Playground'

export default () => (

Grid example:

<Playground>
  {`
    rowStyles = { marginBottom: '20px' };
    columnStyles = { backgroundColor: '#43B02A', border: '1px solid white', textAlign: 'center' };
    <div>
      <Grid style={{ backgroundColor: '#eee' }}>
        <Row style={rowStyles} maxColumns={10}>
          <Column sizes={{ sm: 2 }} style={columnStyles}><p>One</p></Column>
          <Column sizes={{ sm: 2 }} style={columnStyles}><p>Two</p></Column>
          <Column sizes={{ sm: 2 }} style={columnStyles}><p>Three</p></Column>
          <Column sizes={{ sm: 2 }} style={columnStyles}><p>Four</p></Column>
          <Column sizes={{ sm: 2 }} style={columnStyles}><p>Five</p></Column>
          <Column sizes={{ sm: 2 }} style={columnStyles}><p>Six</p></Column>
          <Column sizes={{ sm: 2 }} style={columnStyles}><p>Seven</p></Column>
        </Row>
        <Row style={rowStyles} maxColumns={6}>
          <Column sizes={{ sm: 2 }} style={columnStyles}><p>One</p></Column>
          <Column sizes={{ sm: 2 }} style={columnStyles}><p>Two</p></Column>
          <Column sizes={{ sm: 2 }} style={columnStyles}><p>Three</p></Column>
          <Column sizes={{ sm: 2 }} style={columnStyles}><p>Four</p></Column>
          <Column sizes={{ sm: 2 }} style={columnStyles}><p>Five</p></Column>
          <Column sizes={{ sm: 2 }} style={columnStyles}><p>Six</p></Column>
          <Column sizes={{ sm: 2 }} style={columnStyles}><p>Seven</p></Column>
        </Row>
      </Grid>
    </div>
  `}
</Playground>

)
