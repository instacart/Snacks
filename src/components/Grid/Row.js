import React      from 'react'
import PropTypes  from 'prop-types'
import Radium from 'radium'
import responsive from '../../styles/responsive'
import { supportsCSSGrid } from '../../utils/detectFeature'

// used for browsers without css grid support
const legacyStyles = {
  display: 'block',

  [responsive.sm]: {
    width: `${6 * responsive.columnWidth}px`
  },

  [responsive.md]: {
    width: `${8 * responsive.columnWidth}px`
  },

  [responsive.mdLg]: {
    width: `${10 * responsive.columnWidth}px`
  },

  [responsive.lg]: {
    width: `${12 * responsive.columnWidth}px`
  },

  [responsive.xl]: {
    width: `${14 * responsive.columnWidth}px`
  }
}

const styles = {
  [responsive.xs]: {
    display: 'block'
  },

  [responsive.sm]: {
    display: 'grid',
    gridTemplateColumns: `repeat(6, ${responsive.columnWidth}px)`
  },

  [responsive.md]: {
    display: 'grid',
    gridTemplateColumns: `repeat(8, ${responsive.columnWidth}px)`
  },

  [responsive.mdLg]: {
    display: 'grid',
    gridTemplateColumns: `repeat(10, ${responsive.columnWidth}px)`
  },

  [responsive.lg]: {
    display: 'grid',
    gridTemplateColumns: `repeat(12, ${responsive.columnWidth}px)`
  },

  [responsive.xl]: {
    display: 'grid',
    gridTemplateColumns: `repeat(14, ${responsive.columnWidth}px)`
  }
}

const getMaxColumnsStyles = props => {
  const { maxColumns } = props

  const override = supportsCSSGrid() ? {
    gridTemplateColumns: `repeat(${maxColumns}, ${responsive.columnWidth}px)`,
    justifyContent: 'center'
  } : {
    width: `${maxColumns * responsive.columnWidth}px`,
    marginTop: 0,
    marginRight: 'auto',
    marginBottom: 0,
    marginLeft: 'auto',
  }

  if (maxColumns <= 6) {
    return {
      [responsive.md]: override,
      [responsive.mdLg]: override,
      [responsive.lg]: override,
      [responsive.xl]: override
    }
  } else if (maxColumns <= 8) {
    return {
      [responsive.mdLg]: override,
      [responsive.lg]: override,
      [responsive.xl]: override
    }
  } else if (maxColumns <= 10) {
    return {
      [responsive.lg]:override,
      [responsive.xl]: override
    }
  } else if (maxColumns <= 12) {
    return {
      [responsive.xl]: override
    }
  }

  return {}
}

const fullWidthStyles = ['xs', 'sm', 'md', 'mdLg', 'lg', 'xl'].reduce(
  (styles, size) => ({
    ...styles,
    [responsive[size]]: {
      display: 'block',
      marginTop: `0 calc(-1 * (100vw - ${responsive.screenWidths[size]}px) / 2)`,
      marginRight: `0 calc(-1 * (100vw - ${responsive.screenWidths[size]}px) / 2)`,
      marginBottom: `0 calc(-1 * (100vw - ${responsive.screenWidths[size]}px) / 2)`,
      marginLeft: `0 calc(-1 * (100vw - ${responsive.screenWidths[size]}px) / 2)`
    }
  }),
  { width: '100vw', justifyContent: 'center' }
)

const getFullWidthStyles = props => props.forceFullPage ? fullWidthStyles : {}

const Row = props => {
  const componentStyles = supportsCSSGrid() ? styles : legacyStyles

  return (
    <div
      style={[
        componentStyles,
        getMaxColumnsStyles(props),
        getFullWidthStyles(props),
        props.style
      ]}
    >
      { props.children }
    </div>
  )
}

Row.propTypes = {
  /** Force Row to width of 100vw -- Snacks will add negative margin */
  forceFullPage: PropTypes.bool,
  /** Maximum number of columns this Row should grow to as screen width increases. Cannot exceed 14. */
  maxColumns: PropTypes.number,
  /** Optional style overrides */
  style: PropTypes.object
}

Row.defaultProps = {
  maxColumns: 14
}

export default Radium(Row)
