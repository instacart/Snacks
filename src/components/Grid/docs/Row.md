Grid example:

    rowStyles = { marginBottom: '20px' };
    columnStyles = { backgroundColor: '#43B02A', border: '1px solid white', textAlign: 'center' };
    <div>
      <Grid styles={{ backgroundColor: '#eee' }}>
        <Row styles={rowStyles} maxColumns={10}>
          <Column sizes={{ sm: 2 }} styles={columnStyles}><p>One</p></Column>
          <Column sizes={{ sm: 2 }} styles={columnStyles}><p>Two</p></Column>
          <Column sizes={{ sm: 2 }} styles={columnStyles}><p>Three</p></Column>
          <Column sizes={{ sm: 2 }} styles={columnStyles}><p>Four</p></Column>
          <Column sizes={{ sm: 2 }} styles={columnStyles}><p>Five</p></Column>
          <Column sizes={{ sm: 2 }} styles={columnStyles}><p>Six</p></Column>
          <Column sizes={{ sm: 2 }} styles={columnStyles}><p>Seven</p></Column>
        </Row>
        <Row styles={rowStyles} maxColumns={6}>
          <Column sizes={{ sm: 2 }} styles={columnStyles}><p>One</p></Column>
          <Column sizes={{ sm: 2 }} styles={columnStyles}><p>Two</p></Column>
          <Column sizes={{ sm: 2 }} styles={columnStyles}><p>Three</p></Column>
          <Column sizes={{ sm: 2 }} styles={columnStyles}><p>Four</p></Column>
          <Column sizes={{ sm: 2 }} styles={columnStyles}><p>Five</p></Column>
          <Column sizes={{ sm: 2 }} styles={columnStyles}><p>Six</p></Column>
          <Column sizes={{ sm: 2 }} styles={columnStyles}><p>Seven</p></Column>
        </Row>
        <Row styles={rowStyles} forceFullPage={true}>
          <Column sizes={{ sm: 2 }} styles={columnStyles}><p>One</p></Column>
          <Column sizes={{ sm: 4 }} styles={columnStyles}><p>Two</p></Column>
          <Column sizes={{ sm: 6 }} styles={columnStyles}><p>Three</p></Column>
          <Column sizes={{ sm: 8 }} styles={columnStyles}><p>Four</p></Column>
          <Column sizes={{ sm: 10 }} styles={columnStyles}><p>Five</p></Column>
          <Column sizes={{ sm: 12 }} styles={columnStyles}><p>Six</p></Column>
          <Column sizes={{ sm: 14 }} styles={columnStyles}><p>Seven</p></Column>
        </Row>
      </Grid>
    </div>
