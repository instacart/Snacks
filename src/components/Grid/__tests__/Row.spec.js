import React from 'react'
import renderer from 'react-test-renderer'
import { StyleRoot } from 'radium'
import Row from '../Row'

it('renders Row correctly', () => {
  const tree = renderer
    .create(
      <StyleRoot>
        <div>
          <Row />
        </div>
      </StyleRoot>
    )
    .toJSON()
  expect(tree).toMatchSnapshot()
})

it('renders Row with styles passed in correctly', () => {
  const tree = renderer
    .create(
      <StyleRoot>
        <div>
          <Row style={{ color: '#ccc', width: '100%' }} />
        </div>
      </StyleRoot>
    )
    .toJSON()
  expect(tree).toMatchSnapshot()
})

it('renders Row with maxColumns correctly', () => {
  const tree = renderer
    .create(
      <StyleRoot>
        <div>
          <Row maxColumns={1} />
          <Row maxColumns={2} />
          <Row maxColumns={3} />
          <Row maxColumns={4} />
          <Row maxColumns={5} />
          <Row maxColumns={6} />
          <Row maxColumns={7} />
          <Row maxColumns={8} />
          <Row maxColumns={9} />
          <Row maxColumns={10} />
          <Row maxColumns={11} />
          <Row maxColumns={12} />
          <Row maxColumns={13} />
          <Row maxColumns={14} />
        </div>
      </StyleRoot>
    )
    .toJSON()
  expect(tree).toMatchSnapshot()
})

it('renders Row with styles and maxColumns passed in correctly', () => {
  const tree = renderer
    .create(
      <StyleRoot>
        <div>
          <Row maxColumns={1} style={{ color: '#ccc', width: '100%' }} />
          <Row maxColumns={6} style={{ color: '#ccc', width: '100%' }} />
          <Row maxColumns={10} style={{ color: '#ccc', width: '100%' }} />
        </div>
      </StyleRoot>
    )
    .toJSON()
  expect(tree).toMatchSnapshot()
})

it('renders Row with styles, maxColumns and forceFullPage passed in correctly', () => {
  const tree = renderer
    .create(
      <StyleRoot>
        <div>
          <Row forceFullPage maxColumns={8} />
          <Row forceFullPage style={{ color: '#ccc', width: '100%' }} />
          <Row forceFullPage maxColumns={10} style={{ color: '#ccc', width: '100%' }} />
        </div>
      </StyleRoot>
    )
    .toJSON()
  expect(tree).toMatchSnapshot()
})
