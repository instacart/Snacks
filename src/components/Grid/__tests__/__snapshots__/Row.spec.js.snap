// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders Row correctly 1`] = `
<div
  data-radium={true}
>
  <div>
    <div
      className="rmq-e9fd10a8 rmq-94de0125 rmq-eeb4761a rmq-3e10bf34 rmq-2295bbb9 rmq-108b114"
      data-radium={true}
      style={Object {}}
    />
  </div>
  <style
    dangerouslySetInnerHTML={
      Object {
        "__html": "@media (max-width: 767px){ .rmq-e9fd10a8{display: block !important;}}
@media (min-width: 768px) and (max-width: 831px){ .rmq-94de0125{display: grid !important;
grid-template-columns: repeat(6, 104px) !important;}}
@media (min-width: 832px) and (max-width: 1039px){ .rmq-eeb4761a{display: grid !important;
grid-template-columns: repeat(8, 104px) !important;}}
@media (min-width: 1040px) and (max-width: 1247px){ .rmq-3e10bf34{display: grid !important;
grid-template-columns: repeat(10, 104px) !important;}}
@media (min-width: 1248px) and (max-width: 1455px){ .rmq-2295bbb9{display: grid !important;
grid-template-columns: repeat(12, 104px) !important;}}
@media (min-width: 1456px){ .rmq-108b114{display: grid !important;
grid-template-columns: repeat(14, 104px) !important;}}",
      }
    }
  />
</div>
`;

exports[`renders Row with maxColumns correctly 1`] = `
<div
  data-radium={true}
>
  <div
    data-radium={true}
  >
    <div
      className="rmq-e9fd10a8 rmq-94de0125 rmq-eeb4761a rmq-3e10bf34 rmq-2295bbb9 rmq-108b114 rmq-797f2370 rmq-4560f2a7 rmq-6d474e68 rmq-d80279c3"
      data-radium={true}
      style={Object {}}
    />
    <div
      className="rmq-e9fd10a8 rmq-94de0125 rmq-eeb4761a rmq-3e10bf34 rmq-2295bbb9 rmq-108b114 rmq-d1a52bb3 rmq-164f544 rmq-8beb698b rmq-57e78aa0"
      data-radium={true}
      style={Object {}}
    />
    <div
      className="rmq-e9fd10a8 rmq-94de0125 rmq-eeb4761a rmq-3e10bf34 rmq-2295bbb9 rmq-108b114 rmq-45d5972 rmq-15fad365 rmq-363873aa rmq-d1024a01"
      data-radium={true}
      style={Object {}}
    />
    <div
      className="rmq-e9fd10a8 rmq-94de0125 rmq-eeb4761a rmq-3e10bf34 rmq-2295bbb9 rmq-108b114 rmq-454212b5 rmq-d3dba02 rmq-b91980cd rmq-da5d6e6"
      data-radium={true}
      style={Object {}}
    />
    <div
      className="rmq-e9fd10a8 rmq-94de0125 rmq-eeb4761a rmq-3e10bf34 rmq-2295bbb9 rmq-108b114 rmq-77fa4074 rmq-21d39823 rmq-63668aec rmq-86c09647"
      data-radium={true}
      style={Object {}}
    />
    <div
      className="rmq-e9fd10a8 rmq-94de0125 rmq-eeb4761a rmq-3e10bf34 rmq-2295bbb9 rmq-108b114 rmq-d02048b7 rmq-ddd79ac0 rmq-820aa60f rmq-6a5a724"
      data-radium={true}
      style={Object {}}
    />
    <div
      className="rmq-e9fd10a8 rmq-94de0125 rmq-eeb4761a rmq-3e10bf34 rmq-2295bbb9 rmq-108b114 rmq-f26d78e1 rmq-2c57b02e rmq-7fc06685"
      data-radium={true}
      style={Object {}}
    />
    <div
      className="rmq-e9fd10a8 rmq-94de0125 rmq-eeb4761a rmq-3e10bf34 rmq-2295bbb9 rmq-108b114 rmq-abeb198e rmq-3003f41 rmq-c89db96a"
      data-radium={true}
      style={Object {}}
    />
    <div
      className="rmq-e9fd10a8 rmq-94de0125 rmq-eeb4761a rmq-3e10bf34 rmq-2295bbb9 rmq-108b114 rmq-ad4d4960 rmq-41b878cb"
      data-radium={true}
      style={Object {}}
    />
    <div
      className="rmq-e9fd10a8 rmq-94de0125 rmq-eeb4761a rmq-3e10bf34 rmq-2295bbb9 rmq-108b114 rmq-820d5078 rmq-98718fd3"
      data-radium={true}
      style={Object {}}
    />
    <div
      className="rmq-e9fd10a8 rmq-94de0125 rmq-eeb4761a rmq-3e10bf34 rmq-2295bbb9 rmq-108b114 rmq-692dd1d2"
      data-radium={true}
      style={Object {}}
    />
    <div
      className="rmq-e9fd10a8 rmq-94de0125 rmq-eeb4761a rmq-3e10bf34 rmq-2295bbb9 rmq-108b114 rmq-7ff1dad1"
      data-radium={true}
      style={Object {}}
    />
    <div
      className="rmq-e9fd10a8 rmq-94de0125 rmq-eeb4761a rmq-3e10bf34 rmq-2295bbb9 rmq-108b114"
      data-radium={true}
      style={Object {}}
    />
    <div
      className="rmq-e9fd10a8 rmq-94de0125 rmq-eeb4761a rmq-3e10bf34 rmq-2295bbb9 rmq-108b114"
      data-radium={true}
      style={Object {}}
    />
  </div>
  <style
    dangerouslySetInnerHTML={
      Object {
        "__html": "@media (max-width: 767px){ .rmq-e9fd10a8{display: block !important;}}
@media (min-width: 768px) and (max-width: 831px){ .rmq-94de0125{display: grid !important;
grid-template-columns: repeat(6, 104px) !important;}}
@media (min-width: 832px) and (max-width: 1039px){ .rmq-eeb4761a{display: grid !important;
grid-template-columns: repeat(8, 104px) !important;}}
@media (min-width: 1040px) and (max-width: 1247px){ .rmq-3e10bf34{display: grid !important;
grid-template-columns: repeat(10, 104px) !important;}}
@media (min-width: 1248px) and (max-width: 1455px){ .rmq-2295bbb9{display: grid !important;
grid-template-columns: repeat(12, 104px) !important;}}
@media (min-width: 1456px){ .rmq-108b114{display: grid !important;
grid-template-columns: repeat(14, 104px) !important;}}
@media (min-width: 832px) and (max-width: 1039px) { .rmq-797f2370{grid-template-columns: repeat(1, 104px) !important;
justify-content: center !important;}}
@media (min-width: 1040px) and (max-width: 1247px) { .rmq-4560f2a7{grid-template-columns: repeat(1, 104px) !important;
justify-content: center !important;}}
@media (min-width: 1248px) and (max-width: 1455px) { .rmq-6d474e68{grid-template-columns: repeat(1, 104px) !important;
justify-content: center !important;}}
@media (min-width: 1456px) { .rmq-d80279c3{grid-template-columns: repeat(1, 104px) !important;
justify-content: center !important;}}
@media (min-width: 832px) and (max-width: 1039px) { .rmq-d1a52bb3{grid-template-columns: repeat(2, 104px) !important;
justify-content: center !important;}}
@media (min-width: 1040px) and (max-width: 1247px) { .rmq-164f544{grid-template-columns: repeat(2, 104px) !important;
justify-content: center !important;}}
@media (min-width: 1248px) and (max-width: 1455px) { .rmq-8beb698b{grid-template-columns: repeat(2, 104px) !important;
justify-content: center !important;}}
@media (min-width: 1456px) { .rmq-57e78aa0{grid-template-columns: repeat(2, 104px) !important;
justify-content: center !important;}}
@media (min-width: 832px) and (max-width: 1039px) { .rmq-45d5972{grid-template-columns: repeat(3, 104px) !important;
justify-content: center !important;}}
@media (min-width: 1040px) and (max-width: 1247px) { .rmq-15fad365{grid-template-columns: repeat(3, 104px) !important;
justify-content: center !important;}}
@media (min-width: 1248px) and (max-width: 1455px) { .rmq-363873aa{grid-template-columns: repeat(3, 104px) !important;
justify-content: center !important;}}
@media (min-width: 1456px) { .rmq-d1024a01{grid-template-columns: repeat(3, 104px) !important;
justify-content: center !important;}}
@media (min-width: 832px) and (max-width: 1039px) { .rmq-454212b5{grid-template-columns: repeat(4, 104px) !important;
justify-content: center !important;}}
@media (min-width: 1040px) and (max-width: 1247px) { .rmq-d3dba02{grid-template-columns: repeat(4, 104px) !important;
justify-content: center !important;}}
@media (min-width: 1248px) and (max-width: 1455px) { .rmq-b91980cd{grid-template-columns: repeat(4, 104px) !important;
justify-content: center !important;}}
@media (min-width: 1456px) { .rmq-da5d6e6{grid-template-columns: repeat(4, 104px) !important;
justify-content: center !important;}}
@media (min-width: 832px) and (max-width: 1039px) { .rmq-77fa4074{grid-template-columns: repeat(5, 104px) !important;
justify-content: center !important;}}
@media (min-width: 1040px) and (max-width: 1247px) { .rmq-21d39823{grid-template-columns: repeat(5, 104px) !important;
justify-content: center !important;}}
@media (min-width: 1248px) and (max-width: 1455px) { .rmq-63668aec{grid-template-columns: repeat(5, 104px) !important;
justify-content: center !important;}}
@media (min-width: 1456px) { .rmq-86c09647{grid-template-columns: repeat(5, 104px) !important;
justify-content: center !important;}}
@media (min-width: 832px) and (max-width: 1039px) { .rmq-d02048b7{grid-template-columns: repeat(6, 104px) !important;
justify-content: center !important;}}
@media (min-width: 1040px) and (max-width: 1247px) { .rmq-ddd79ac0{grid-template-columns: repeat(6, 104px) !important;
justify-content: center !important;}}
@media (min-width: 1248px) and (max-width: 1455px) { .rmq-820aa60f{grid-template-columns: repeat(6, 104px) !important;
justify-content: center !important;}}
@media (min-width: 1456px) { .rmq-6a5a724{grid-template-columns: repeat(6, 104px) !important;
justify-content: center !important;}}
@media (min-width: 1040px) and (max-width: 1247px) { .rmq-f26d78e1{grid-template-columns: repeat(7, 104px) !important;
justify-content: center !important;}}
@media (min-width: 1248px) and (max-width: 1455px) { .rmq-2c57b02e{grid-template-columns: repeat(7, 104px) !important;
justify-content: center !important;}}
@media (min-width: 1456px) { .rmq-7fc06685{grid-template-columns: repeat(7, 104px) !important;
justify-content: center !important;}}
@media (min-width: 1040px) and (max-width: 1247px) { .rmq-abeb198e{grid-template-columns: repeat(8, 104px) !important;
justify-content: center !important;}}
@media (min-width: 1248px) and (max-width: 1455px) { .rmq-3003f41{grid-template-columns: repeat(8, 104px) !important;
justify-content: center !important;}}
@media (min-width: 1456px) { .rmq-c89db96a{grid-template-columns: repeat(8, 104px) !important;
justify-content: center !important;}}
@media (min-width: 1248px) and (max-width: 1455px) { .rmq-ad4d4960{grid-template-columns: repeat(9, 104px) !important;
justify-content: center !important;}}
@media (min-width: 1456px) { .rmq-41b878cb{grid-template-columns: repeat(9, 104px) !important;
justify-content: center !important;}}
@media (min-width: 1248px) and (max-width: 1455px) { .rmq-820d5078{grid-template-columns: repeat(10, 104px) !important;
justify-content: center !important;}}
@media (min-width: 1456px) { .rmq-98718fd3{grid-template-columns: repeat(10, 104px) !important;
justify-content: center !important;}}
@media (min-width: 1456px) { .rmq-692dd1d2{grid-template-columns: repeat(11, 104px) !important;
justify-content: center !important;}}
@media (min-width: 1456px) { .rmq-7ff1dad1{grid-template-columns: repeat(12, 104px) !important;
justify-content: center !important;}}",
      }
    }
  />
</div>
`;

exports[`renders Row with styles and maxColumns passed in correctly 1`] = `
<div
  data-radium={true}
>
  <div
    data-radium={true}
  >
    <div
      className="rmq-e9fd10a8 rmq-94de0125 rmq-eeb4761a rmq-3e10bf34 rmq-2295bbb9 rmq-108b114 rmq-797f2370 rmq-4560f2a7 rmq-6d474e68 rmq-d80279c3"
      data-radium={true}
      style={
        Object {
          "color": "#ccc",
          "width": "100%",
        }
      }
    />
    <div
      className="rmq-e9fd10a8 rmq-94de0125 rmq-eeb4761a rmq-3e10bf34 rmq-2295bbb9 rmq-108b114 rmq-d02048b7 rmq-ddd79ac0 rmq-820aa60f rmq-6a5a724"
      data-radium={true}
      style={
        Object {
          "color": "#ccc",
          "width": "100%",
        }
      }
    />
    <div
      className="rmq-e9fd10a8 rmq-94de0125 rmq-eeb4761a rmq-3e10bf34 rmq-2295bbb9 rmq-108b114 rmq-820d5078 rmq-98718fd3"
      data-radium={true}
      style={
        Object {
          "color": "#ccc",
          "width": "100%",
        }
      }
    />
  </div>
  <style
    dangerouslySetInnerHTML={
      Object {
        "__html": "@media (max-width: 767px){ .rmq-e9fd10a8{display: block !important;}}
@media (min-width: 768px) and (max-width: 831px){ .rmq-94de0125{display: grid !important;
grid-template-columns: repeat(6, 104px) !important;}}
@media (min-width: 832px) and (max-width: 1039px){ .rmq-eeb4761a{display: grid !important;
grid-template-columns: repeat(8, 104px) !important;}}
@media (min-width: 1040px) and (max-width: 1247px){ .rmq-3e10bf34{display: grid !important;
grid-template-columns: repeat(10, 104px) !important;}}
@media (min-width: 1248px) and (max-width: 1455px){ .rmq-2295bbb9{display: grid !important;
grid-template-columns: repeat(12, 104px) !important;}}
@media (min-width: 1456px){ .rmq-108b114{display: grid !important;
grid-template-columns: repeat(14, 104px) !important;}}
@media (min-width: 832px) and (max-width: 1039px) { .rmq-797f2370{grid-template-columns: repeat(1, 104px) !important;
justify-content: center !important;}}
@media (min-width: 1040px) and (max-width: 1247px) { .rmq-4560f2a7{grid-template-columns: repeat(1, 104px) !important;
justify-content: center !important;}}
@media (min-width: 1248px) and (max-width: 1455px) { .rmq-6d474e68{grid-template-columns: repeat(1, 104px) !important;
justify-content: center !important;}}
@media (min-width: 1456px) { .rmq-d80279c3{grid-template-columns: repeat(1, 104px) !important;
justify-content: center !important;}}
@media (min-width: 832px) and (max-width: 1039px) { .rmq-d02048b7{grid-template-columns: repeat(6, 104px) !important;
justify-content: center !important;}}
@media (min-width: 1040px) and (max-width: 1247px) { .rmq-ddd79ac0{grid-template-columns: repeat(6, 104px) !important;
justify-content: center !important;}}
@media (min-width: 1248px) and (max-width: 1455px) { .rmq-820aa60f{grid-template-columns: repeat(6, 104px) !important;
justify-content: center !important;}}
@media (min-width: 1456px) { .rmq-6a5a724{grid-template-columns: repeat(6, 104px) !important;
justify-content: center !important;}}
@media (min-width: 1248px) and (max-width: 1455px) { .rmq-820d5078{grid-template-columns: repeat(10, 104px) !important;
justify-content: center !important;}}
@media (min-width: 1456px) { .rmq-98718fd3{grid-template-columns: repeat(10, 104px) !important;
justify-content: center !important;}}",
      }
    }
  />
</div>
`;

exports[`renders Row with styles passed in correctly 1`] = `
<div
  data-radium={true}
>
  <div>
    <div
      className="rmq-e9fd10a8 rmq-94de0125 rmq-eeb4761a rmq-3e10bf34 rmq-2295bbb9 rmq-108b114"
      data-radium={true}
      style={
        Object {
          "color": "#ccc",
          "width": "100%",
        }
      }
    />
  </div>
  <style
    dangerouslySetInnerHTML={
      Object {
        "__html": "@media (max-width: 767px){ .rmq-e9fd10a8{display: block !important;}}
@media (min-width: 768px) and (max-width: 831px){ .rmq-94de0125{display: grid !important;
grid-template-columns: repeat(6, 104px) !important;}}
@media (min-width: 832px) and (max-width: 1039px){ .rmq-eeb4761a{display: grid !important;
grid-template-columns: repeat(8, 104px) !important;}}
@media (min-width: 1040px) and (max-width: 1247px){ .rmq-3e10bf34{display: grid !important;
grid-template-columns: repeat(10, 104px) !important;}}
@media (min-width: 1248px) and (max-width: 1455px){ .rmq-2295bbb9{display: grid !important;
grid-template-columns: repeat(12, 104px) !important;}}
@media (min-width: 1456px){ .rmq-108b114{display: grid !important;
grid-template-columns: repeat(14, 104px) !important;}}",
      }
    }
  />
</div>
`;

exports[`renders Row with styles, maxColumns and forceFullPage passed in correctly 1`] = `
<div
  data-radium={true}
>
  <div
    data-radium={true}
  >
    <div
      className="rmq-e9fd10a8 rmq-94de0125 rmq-eeb4761a rmq-3e10bf34 rmq-2295bbb9 rmq-108b114 rmq-abeb198e rmq-3003f41 rmq-c89db96a rmq-da8cd0e3 rmq-3eba0fa5 rmq-4794023d rmq-3cc881d6 rmq-902176d3 rmq-9a0c63b1"
      data-radium={true}
      style={
        Object {
          "justifyContent": "center",
          "width": "100vw",
        }
      }
    />
    <div
      className="rmq-e9fd10a8 rmq-94de0125 rmq-eeb4761a rmq-3e10bf34 rmq-2295bbb9 rmq-108b114 rmq-da8cd0e3 rmq-3eba0fa5 rmq-4794023d rmq-90efeef6 rmq-7d240233 rmq-49dba7f1"
      data-radium={true}
      style={
        Object {
          "color": "#ccc",
          "justifyContent": "center",
          "width": "100%",
        }
      }
    />
    <div
      className="rmq-e9fd10a8 rmq-94de0125 rmq-eeb4761a rmq-3e10bf34 rmq-2295bbb9 rmq-108b114 rmq-820d5078 rmq-98718fd3 rmq-da8cd0e3 rmq-3eba0fa5 rmq-4794023d rmq-90efeef6 rmq-902176d3 rmq-9a0c63b1"
      data-radium={true}
      style={
        Object {
          "color": "#ccc",
          "justifyContent": "center",
          "width": "100%",
        }
      }
    />
  </div>
  <style
    dangerouslySetInnerHTML={
      Object {
        "__html": "@media (max-width: 767px){ .rmq-e9fd10a8{display: block !important;}}
@media (min-width: 768px) and (max-width: 831px){ .rmq-94de0125{display: grid !important;
grid-template-columns: repeat(6, 104px) !important;}}
@media (min-width: 832px) and (max-width: 1039px){ .rmq-eeb4761a{display: grid !important;
grid-template-columns: repeat(8, 104px) !important;}}
@media (min-width: 1040px) and (max-width: 1247px){ .rmq-3e10bf34{display: grid !important;
grid-template-columns: repeat(10, 104px) !important;}}
@media (min-width: 1248px) and (max-width: 1455px){ .rmq-2295bbb9{display: grid !important;
grid-template-columns: repeat(12, 104px) !important;}}
@media (min-width: 1456px){ .rmq-108b114{display: grid !important;
grid-template-columns: repeat(14, 104px) !important;}}
@media (min-width: 1040px) and (max-width: 1247px) { .rmq-abeb198e{grid-template-columns: repeat(8, 104px) !important;
justify-content: center !important;}}
@media (min-width: 1248px) and (max-width: 1455px) { .rmq-3003f41{grid-template-columns: repeat(8, 104px) !important;
justify-content: center !important;}}
@media (min-width: 1456px) { .rmq-c89db96a{grid-template-columns: repeat(8, 104px) !important;
justify-content: center !important;}}
@media (max-width: 767px) { .rmq-da8cd0e3{display: block !important;
margin: 0 calc(-1 * (100vw - undefinedpx) / 2) !important;}}
@media (min-width: 768px) and (max-width: 831px) { .rmq-3eba0fa5{display: block !important;
margin: 0 calc(-1 * (100vw - 624px) / 2) !important;}}
@media (min-width: 832px) and (max-width: 1039px) { .rmq-4794023d{display: block !important;
margin: 0 calc(-1 * (100vw - 832px) / 2) !important;}}
@media (min-width: 1040px) and (max-width: 1247px)  { .rmq-3cc881d6{display: block !important;
margin: 0 calc(-1 * (100vw - 1040px) / 2) !important;}}
@media (min-width: 1248px) and (max-width: 1455px)  { .rmq-902176d3{display: block !important;
margin: 0 calc(-1 * (100vw - 1248px) / 2) !important;}}
@media (min-width: 1456px)  { .rmq-9a0c63b1{display: block !important;
margin: 0 calc(-1 * (100vw - 1456px) / 2) !important;}}
@media (min-width: 1040px) and (max-width: 1247px) { .rmq-90efeef6{display: block !important;
margin: 0 calc(-1 * (100vw - 1040px) / 2) !important;}}
@media (min-width: 1248px) and (max-width: 1455px) { .rmq-7d240233{display: block !important;
margin: 0 calc(-1 * (100vw - 1248px) / 2) !important;}}
@media (min-width: 1456px) { .rmq-49dba7f1{display: block !important;
margin: 0 calc(-1 * (100vw - 1456px) / 2) !important;}}
@media (min-width: 1248px) and (max-width: 1455px) { .rmq-820d5078{grid-template-columns: repeat(10, 104px) !important;
justify-content: center !important;}}
@media (min-width: 1456px) { .rmq-98718fd3{grid-template-columns: repeat(10, 104px) !important;
justify-content: center !important;}}",
      }
    }
  />
</div>
`;
